src/chainparams.cpp

```diff
@@ -284,7 +284,7 @@ class CMainParams : public CChainParams {
         fRequireRoutableExternalIP = true;
         m_is_test_chain = false;
         fAllowMultipleAddressesFromGroup = false;
-        fAllowMultiplePorts = false;
+        fAllowMultiplePorts = true;
```

Comment:
6451dd7ffa7cf88c1dcfb3f1d159a34401db56d4:

> allow some mainnet peers to use non-default ports

It's already allowed for non-masternodes, just use `-port`.

`fAllowMultiplePorts` was introduced in https://github.com/dashpay/dash/pull/1967 to preserve the behaviour of the old non-deterministic masternode list while fixing `netfulfildman`. Most of that code is gone and `fAllowMultiplePorts` is kind of useless (and even confusing). Should be safe to drop it completely imo. Would prefer this to be done in a separate PR though.

Reply:
Pls see #6538. Found a bunch of other potential issues while looking at it...

Reply:
Removed 6451dd7ffa7cf88c1dcfb3f1d159a34401db56d4 from PR as [dash#6538](https://github.com/dashpay/dash/pull/6538) is more fleshed out, resolved in latest push!